### ISSUE TITLE
Incorrect dynamic context emitted for dynamic calls in async lambdas

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#2720]
```
using System;
using System.Threading.Tasks;

class C
{
    static void Main()
    {
        dynamic d = Task.FromResult("a");
        G(async () => await d());
    }

    static void G(Func<Task<object>> f)
    {
    }
}
```

Dev12 has ```ldtoken C``` whereas Roslyn produces ```ldtoken C/'<>c__DisplayClass0_0'/'<<Main>b__0>d'```.

**Comment by amcasey** [comment]
A fix will probably fall out of some EE work I'm doing.

**Comment by amcasey** [comment]
This was fixed by a38e55fa0a8f710ad67f5834751181a6db5ce27d.
